### ISSUE TITLE
Periodically log stack traces

### ISSUE THREAD

**Issue by breeswish** [pingcap/tikv#3083]
Currently when we find out that TiKV was using unreasonable high CPU resources, we can only try to infer the cause by looking at metrics and logs. However sometimes this won't work , for example, library bugs. Flamegraphs won't work as well for non-reproduceble issues.

If we can trace the application stack periodically, with a not-too-small interval (let's say, 1 minute), we may able to locate the root cause of such issues much more easily, without significantly affect the application performance.

**Comment by siddontang** [comment]
you can use systemtap to do it directly.

**Comment by breeswish** [comment]
Yes, but it would be better to do it in userland, i.e. without special privileges, so that it can be always carried with our application.

**Comment by BusyJay** [comment]
It feels odd to implement such feature. Is it a common behavior that other similar products also implement? Can pstack solve your problem?

**Comment by breeswish** [comment]
Actually TiDB directly provides a feature to grab current stack traces. However after investigation it seems to be difficult to implement this feature inside TiKV. pstack should be a nice replacement. We can implement it in ansible.
